### PR TITLE
tag `get_inputs` span with `iteration`

### PR DESCRIPTION
**Pull request by danielnorberg**
Add `iteration` tag to `get_inputs` span so we can search for slow initial `get_inputs` spans in cold-start traces

### CODE REVIEW

File: modal/_container_entrypoint.py [hunk 1]
```diff
@@ -178,11 +179,15 @@ async def _generate_inputs(
     ) -> AsyncIterator[tuple[str, api_pb2.FunctionInput]]:
         request = api_pb2.FunctionGetInputsRequest(function_id=self.function_id)
         eof_received = False
+        iteration = 0
         while not eof_received:
             request.average_call_time = self.get_average_call_time()
             request.max_values = self.get_max_inputs_to_fetch()  # Deprecated; remove.
 
             with trace("get_inputs"):
+                span = tracer.current_span()
```
Comment (freider) [inline modal/_container_entrypoint.py:188]: Not sure how ddtrace works, but I assume this is safe to call even when the tracing is disabled?

Comment (danielnorberg) [reply modal/_container_entrypoint.py:188]: derp nej, inte i python 👎

**Comment by danielnorberg** [comment]
tests are failing with 
> ModuleNotFoundError: No module named 'ddtrace'
🤔

**Comment by danielnorberg** [comment]
oh, `ddtrace` might not be present if tracing is disabled

File: modal/_container_entrypoint.py (hunk 1 above)
Comment (danielnorberg) [reply modal/_container_entrypoint.py]: ptal again

**Review by freider — APPROVED** [review]

**Pull request merged**